packages/babel-parser/src/util/whitespace.js

```diff
@@ -1,13 +1,53 @@
 // @flow
 
+import * as charCodes from "charcodes";
+
 // Matches a whole line break (where CRLF is considered a single
 // line break). Used to count lines.
-
 export const lineBreak = /\r\n?|\n|\u2028|\u2029/;
 export const lineBreakG = new RegExp(lineBreak.source, "g");
 
+// https://tc39.github.io/ecma262/#sec-line-terminators
 export function isNewLine(code: number): boolean {
-  return code === 10 || code === 13 || code === 0x2028 || code === 0x2029;
+  switch (code) {
+    case charCodes.lineFeed:
```

Comment:
Out of curiosity, does using properties like this cause V8 to optimize this less? In theory it'd now have to access the property ever time `isNewLine` is called.

Reply:
It does, but we use https://github.com/xtuc/charcodes/tree/master/packages/babel-plugin-transform-charcodes to inline the code.

Reply:
Ahhhh perfect.